src/lib/components/ThemeToggle.svelte

```diff
@@ -1,54 +1,40 @@
 <script lang="ts">
-  import Toggle from "./Toggle.svelte";
   import { Theme } from "$lib/types/theme";
   import { themeStore } from "$lib/stores/theme.store";
   import { i18n } from "$lib/stores/i18n";
   import IconLightMode from "$lib/icons/IconLightMode.svelte";
   import IconDarkMode from "$lib/icons/IconDarkMode.svelte";
 
-  const switchTheme = ({ detail }: CustomEvent<boolean>) =>
-    themeStore.select(detail ? Theme.DARK : Theme.LIGHT);
+  const switchTheme = () => {
+    themeStore.select($themeStore === Theme.LIGHT ? Theme.DARK : Theme.LIGHT);
+  };
 
-  let checked: boolean;
-  $: checked = $themeStore !== Theme.LIGHT;
+  $: isDarkMode = $themeStore === Theme.DARK;
 </script>
 
-<div class="theme-toggle" data-tid="theme-toggle">
-  <div class="toggle">
+<button
+  data-tid="theme-toggle"
+  class="theme-toggle icon-only"
+  on:click={switchTheme}
+  aria-label={$i18n.theme.switch_theme}
+>
+  {#if isDarkMode}
     <IconLightMode />
-    <Toggle
-      bind:checked
-      on:nnsToggle={switchTheme}
-      ariaLabel={$i18n.theme.switch_theme}
-    />
+  {:else}
     <IconDarkMode />
-  </div>
-</div>
+  {/if}
+</button>
 
 <style lang="scss">
   .theme-toggle {
-    display: flex;
-    align-items: center;
-
-    font-size: var(--font-size-h4);
-    line-height: var(--line-height-h4);
-
-    padding: 0 var(--padding-0_5x);
-    gap: var(--padding);
-  }
-
-  .toggle {
-    display: flex;
-    align-items: center;
-    grid-template-columns: repeat(3, auto);
-    grid-column-gap: 2px;
-
-    :global(svg) {
-      width: var(--padding-2x);
-      height: var(--padding-2x);
-
-      &:first-of-type {
-        margin-right: calc(var(--padding-0_5x) / 2);
+    color: var(--text-color);
+    padding: var(--padding);
+    background: var(--sidebar-button-background);
+
+    &:hover {
+      background: var(--sidebar-button-background-hover);
+      :global(svg) {
```

Comment:
The other option is to track the hover with js save it in a state use different icons for filled and outlined.

I thought this would be a better approach but happy to change it. Also let me know if there is a better way to conditionally render icons based on hover.

Ex: I can put two icons in each if block and based on hover i can change the displays if that is something you would prefer, but i think it is harder to read.

Reply:
The svg can use a CSS variable which you assign on hover.

Reply:
I can for sure do that or modify svgs to take in a fill property, but i thought we would like to be more consistent with how we keep svgs. If some svgs support fill and others don't then it could create a problem.

Reply:
Using a CSS variables seems less prone to issue to me than overriding global but, not a strong opinion neither. If it's fine for the colleagues, fine for me.

Reply:
Added CSS variable

Reply:
I would have expected multiple icons (with different display on hover) but a CSS variable in the icon sounds good to me.
We can add CSS variables to more icons as we need.
It's also fine with me if you want to update all the icons to support a fill variable (in another PR) but not necessary.